digdag-standards/src/main/java/io/digdag/standards/operator/state/PollingWaiter.java

```diff
@@ -102,7 +149,7 @@ public PollingWaiter withPollInterval(DurationInterval retryInterval)
         if (!result.isPresent()) {
             int iteration = pollState.params().get(ITERATION, int.class, 0);
             pollState.params().set(ITERATION, iteration + 1);
-            int interval = (int) Math.min(pollInterval.min().getSeconds() * Math.pow(2, iteration), pollInterval.max().getSeconds());
+            int interval = (int) Math.min(pollInterval.min().getSeconds() * Math.pow(nextIntervalExpBase, iteration), pollInterval.max().getSeconds());
```

Comment:
With exp=1.2, timeout is:
1s, 1.2s, 1.44s, 1.73, ..., 197s (i=29), 237s (i=30)
And cumulative waiting time is:
1s, 2.2s, 3.64s, ..., 19.7m (i=29), 23.7m (i=30)

If user sets 20 minute of timeout, s3_waits waits 23.7 minute, where it should fail after 20 minutes. 3 minutes, 15% of difference is a significant & unexpected.
If we shorten `nextIntervalExpBase`, s3_wait makes too many polling. This is a trade-off.

But if we modify this `interval` based remaining timeout wait, we don't have have the trade-off. If remaining time is 0.3 minute, wait 0.3 minute instead of waiting for 5 minute. This implementation seems easy. I don't think it's good idea to leave such easy improvement.

Please add it, and remove this change https://github.com/treasure-data/digdag/pull/1280/files#diff-5371699d0c423b5e397f66775898325fR48 which is making negative impact.

Reply:
I agree with the this implementation has still impact.
>If remaining time is 0.3 minute, wait 0.3 minute instead of waiting for 5 minute.
I will reconsider based on your idea. In this case users need to wait for until last minute of timeout. In addition, we need to consider the next task will not start precisely, it will be delayed.

Reply:
Next interval should be 55s at the case.
Please also consider jitter though.

Reply:
How about the following way.
Is this acceptable for you?

**interval = max(5sec,  min( (now- start time), (5sec * 2^N), 5min )**

ex1) timeout: 60s
N=0  : interval = 5s       ( max(5s, min(60s, 5s, 5m)) )
N=1  : interval =  10s     ( max(5s, min(55s, 10s, 5m)) )
N=2  : interval =  20s     ( max(5s, min(45s, 20s, 5m)) )
N=3  : interval =  40s     ( max(5s, min(25s, 40s, 5m)) )
N=4  : interval =  5s       ( max(5s, min(5s, 80s, 5m)) )
N=5  : last check then timeout

ex2) timeout: 60m(3600s)
N=0  : interval = 5s       ( max(5s, min(3600s, 5s, 5m)) )
N=1  : interval =  10s     ( max(5s, min(3555s, 10s, 5m)) )
N=2  : interval =  20s     ( max(5s, min(3545s, 20s, 5m)) )
N=3  : interval =  40s     ( max(5s, min(3525s, 40s, 5m)) )
N=4  : interval =  80s     ( max(5s, min(3485s, 80s, 5m)) )
N=5  : interval =  160s   ( max(5s, min(3405s, 160s, 5m)) )
N=6  : interval =  5m     ( max(5s, min(3245s, 320s, 5m)) )
N=7  : interval =  5m     ( max(5s, min(2945s, 320s, 5m)) )
....
N=15  : interval = 5m      ( max(5s, min(545s, 320s, 5m)) )
N=16  : interval =  245s  ( max(5s, min(245s, larger than 5m, 5m)) )
N=17  : last check then timeout

Reply:
After discussion, I will modify as follows:
- max(5s, min(timeout - (now - start time), 5min)   (if timeout set)
- original formula (if timeout not set)